app/substitution_helper.rb

```diff
@@ -10,7 +10,12 @@ def clear!(new_item_document_type:, new_item_content_id:, base_path:, locale:, s
         allowed_to_substitute = (substitute?(new_item_document_type) || substitute?(blocking_item.document_type))
 
         if mismatch && allowed_to_substitute
-          State.substitute(blocking_item)
+          if state == "draft"
+            Commands::V2::DiscardDraft.call(content_id: blocking_item.content_id)
```

Comment:
Hopefully not; the DiscardDraft command uses the same `after_transaction_commit` block that the other commands do for talking downstream.